str0m/src/media/mod.rs

```diff
@@ -1193,6 +1198,31 @@ impl Media {
     pub(crate) fn equalize_requires_ssrcs(&self) -> usize {
         (self.sources_tx.len() as isize - self.sources_rx.len() as isize).unsigned_abs()
     }
+
+    pub fn visit_stats(&self, snapshot: &mut StatsSnapshot) {
+        for s in &self.sources_rx {
+            let bytes_rx = s.get_bytes_received();
+            snapshot.rx += bytes_rx;
+            let key = (self.mid, s.rid());
+            if let Some(v) = snapshot.ingress.get_mut(&key) {
+                *v += bytes_rx;
+            } else {
+                snapshot.ingress.insert(key, bytes_rx);
+            }
+            // TODO: handle remote stats for this ingress source
+        }
+        for s in &self.sources_tx {
+            let bytes_tx = s.get_bytes_sent();
+            snapshot.tx += bytes_tx;
+            let key = (self.mid, s.rid());
+            if let Some(v) = snapshot.ingress.get_mut(&key) {
+                *v += bytes_tx;
+            } else {
+                snapshot.ingress.insert(key, bytes_tx);
+            }
+            // TODO: handle remote stats for this egress source
+        }
+    }
```

Comment:
Wouldn't it be cleaner to implement `visit_stats` on ReceiverSource and SenderSource here?

Reply:
Yeah, pass args in `visit_stats`. I like that.